qubesmanager/utils.py

```diff
@@ -54,6 +54,8 @@ def prepare_choice(widget, holder, propname, choice, default,
 
     if propname is not None:
         oldvalue = getattr(holder, propname)
+        if transform is not None:
+            oldvalue = transform(oldvalue)
         is_default = holder.property_is_default(propname)
```

Comment:
(related to this place, not exactly the above line) IMO this would be a good place for holder.property_get_default() call.